src/libp2p.js

```diff
@@ -253,8 +255,8 @@ class DaemonLibp2p extends Libp2p {
   constructor (libp2pOpts, { announceAddrs }) {
     super(libp2pOpts)
     this.announceAddrs = announceAddrs
-    this.needsPullStream = libp2pOpts.config.EXPERIMENTAL.pubsub
-    this._pubsub = this.pubsub
+    this.needsPullStream = libp2pOpts.config.pubsub.enabled
+    this._pubSub = this.pubsub
```

Comment:
Any particular reason for the change to `_pubSub`? I assume you're trying to avoid collision with `_pubsub` that already exists on libp2p, but it shouldn't be a problem.

Reply:
I renamed it because the api from `js-libp2p` (https://github.com/libp2p/js-libp2p/pull/365/files#diff-1fdf421c05c1140f6d71444ea2b27638R128) gets overwritten and functions for start and stop don't exist anymore

```
this._pubsub.start is not a function
```

Reply:
We should just not set this at all, and only set `this.pubsub = new Pubsub(this)`. I left a comment at https://github.com/libp2p/js-libp2p/pull/365/files#r302888107 that I think could make this a bit easier to follow

Reply:
Done